pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationManager.java

```diff
@@ -150,8 +153,46 @@ public boolean canLookup(DestinationName destination, String role) throws Except
         return canProduce(destination, role) || canConsume(destination, role, null);
     }
 
-    private CompletableFuture<Boolean> checkAuthorization(DestinationName destination, String role,
-            AuthAction action) {
+    /**
+     * Check whether the specified role can perform a lookup for the specified destination.
+     *
+     * For that the caller needs to have producer or consumer permission.
+     *
+     * @param destination
+     * @param role
+     * @return
+     * @throws Exception
+     */
+    public CompletableFuture<Boolean> canLookupAsync(DestinationName destination, String role) {
+        CompletableFuture<Boolean> finalResult = new CompletableFuture<Boolean>();
+        canProduceAsync(destination, role).whenComplete((produceAuthorized, ex) -> {
+            if (ex == null) {
+                if (produceAuthorized) {
+                    finalResult.complete(produceAuthorized);
+                    return;
+                }
+            } else if (log.isDebugEnabled()) {
+                log.debug("Destination [{}] Role [{}] exception occured while trying to check Produce permissions. {}",
+                        destination.toString(), role, ex.getMessage());
+            }
+            canConsumeAsync(destination, role, null).whenComplete((consumeAuthorized, e) -> {
+                if (e == null) {
+                    if (consumeAuthorized) {
+                        finalResult.complete(consumeAuthorized);
+                        return;
+                    }
+                } else if (log.isDebugEnabled()) {
+                    log.debug(
```

Comment:
`canProduceAsync()` and `canConsumeAsync()` fails the future when it has zk-connection issue and in that case, we should give 5XX error instead 403. So, I think we should fail `finalResult` future here instead returning completing it with false.